connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -280,15 +282,16 @@ private void commitTransaction() {
 
         // Commit the transaction
         // Blocks until all outstanding records have been sent and ack'd
-        try {
-            producer.commitTransaction();
-        } catch (Throwable t) {
-            log.error("{} Failed to commit producer transaction", ExactlyOnceWorkerSourceTask.this, t);
-            flushError.compareAndSet(null, t);
+        if (flushFuture != null) {
```

Comment:
This might mean that we never end up doing a commit if there are no offsets to commit, for example if the connector never returns any offsets. There might still be SourceRecords to send. Not sure if any connector does this, but we should not ignore this edge case regardless. 

Maybe we can simply check if flushError is set instead. The callback should be invoked and the error set immediately in case of a serialisation failure.

Reply:
Started with the same thought, but tried making it consistent with [WorkerSourceTask](https://github.com/apache/kafka/blob/3ae1afa43838066e44ea78918050c6780c208042/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java#L253-L267). 

If there are no offsets and no error, we should still commit (producer and task). Need to make changes in WorkerSourceTask as well! 
I'll make the suggested change ! 
thanks!